packages/langium/src/utils/ast-util.ts

```diff
@@ -33,6 +32,17 @@ export function getContainerOfType<T extends AstNode>(node: AstNode | undefined,
     return undefined;
 }
 
+export function hasContainerOfType<T extends AstNode>(node: AstNode | undefined, typePredicate: (n: AstNode) => n is T): boolean {
```

Comment:
I fell like it doesn't make a lot of sense to include the generic parameter in here:
```suggestion
export function hasContainerOfType(node: AstNode | undefined, typePredicate: (n: AstNode) => boolean): boolean {
```

Reply:
But then the type predicate has the type `(n: AstNode) => n is AstNode`, which looks strange, and the compiler complains that line 41 (the _else_ case of `if (typePredicate(item))`) is not reachable.